explore/src/main/scala/explore/observationtree/ConstraintGroupObsList.scala

```diff
@@ -189,16 +198,21 @@ object ConstraintGroupObsList {
                           )
                           .orEmpty
                       )(^.cursor.pointer,
-                        ^.onClick --> props.selected.set(
-                          SelectedPanel.editor(constraintGroup.obsIds)
-                        )
+                        ^.onClick --> {
+                          props.focused.mod(unfocusIfObs) >>
+                            props.selected.set(SelectedPanel.editor(constraintGroup.obsIds))
+                        }
                       )(
                         csHeader,
                         TagMod.when(props.expandedIds.get.contains(obsIds))(
                           cgObs.zipWithIndex.toTagMod { case (obs, idx) =>
-                            props.renderObsBadgeItem(selectable = false,
-                                                     highlightSelected = false,
-                                                     linkToObsTab = true
+                            props.renderObsBadgeItem(
+                              selectable = true,
+                              highlightSelected = true,
+                              forceHighlight = groupSelected && !singleObsSelected,
+                              linkToObsTab = false,
```

Comment:
We could keep `linkToObsTab`. That triggers on double click, right?

Reply:
We could, but I was worried about inadvertent double clicks sending the user off somewhere. It doesn't seem as necessary if they can edit the constraint set for the observation here. If we want to keep a link to the observation, I think it might be nicer to make the observation id in the upper right of the badge be a link. That would be more obvious. With the double click, there isn't any indication that is going to happen.